--- a/configure.ac	2010-04-26 16:38:51.000000000 +0200
+++ b/configure.ac.new	2010-04-28 15:29:04.175108608 +0200
@@ -19,7 +19,7 @@
 m4_define(empathy_version, empathy_full_version[]empathy_maybe_datestamp)
 
 
-AC_INIT(Empathy, empathy_version, http://bugzilla.gnome.org/browse.cgi?product=empathy)
+AC_INIT([Empathy], [empathy_version], [http://bugzilla.gnome.org/browse.cgi?product=empathy])
 
 
 AC_PREREQ(2.59)
