diff -ur empathy-0.12-o/python/pyempathy/Makefile.am empathy-0.12/python/pyempathy/Makefile.am
--- empathy-0.12-o/python/pyempathy/Makefile.am	2007-08-24 15:29:46.000000000 -0600
+++ empathy-0.12/python/pyempathy/Makefile.am	2007-08-25 16:42:30.000000000 -0600
@@ -6,7 +6,7 @@
 	-I$(top_srcdir)                                 \
 	-DDATADIR=\""$(datadir)"\"              	\
 	$(PYTHON_BINDING_CFLAGS)			\
-	`python-config --cflags`			\
+	`python2.5-config --cflags`			\
 	$(WARN_CFLAGS)
 
 BUILT_SOURCES = 					\
@@ -24,7 +24,7 @@
 	$(top_builddir)/libempathy/libempathy.la
 
 empathy_la_LDFLAGS =					\
-	`python-config --libs`				\
+	`python2.5-config --libs`				\
 	-module -avoid-version
 
 pyempathy.c: pyempathy.override pyempathy.defs
diff -ur empathy-0.12-o/python/pyempathygtk/Makefile.am empathy-0.12/python/pyempathygtk/Makefile.am
--- empathy-0.12-o/python/pyempathygtk/Makefile.am	2007-08-24 15:29:46.000000000 -0600
+++ empathy-0.12/python/pyempathygtk/Makefile.am	2007-08-25 16:43:02.000000000 -0600
@@ -6,7 +6,7 @@
 	-I$(top_srcdir)                                 \
 	-DDATADIR=\""$(datadir)"\"              	\
 	$(PYTHON_BINDING_CFLAGS)			\
-	`python-config --cflags`			
+	`python2.5-config --cflags`			
 
 BUILT_SOURCES = 					\
 	pyempathygtk.c					
@@ -24,7 +24,7 @@
 	$(top_builddir)/libempathy-gtk/libempathy-gtk.la
 
 empathygtk_la_LDFLAGS =					\
-	`python-config --libs`				\
+	`python2.5-config --libs`				\
 	-module -avoid-version
 
 pyempathygtk.c: pyempathygtk.override pyempathygtk.defs
