diff --git a/configure.ac b/configure.ac
index 682f6a7..0088b4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,7 @@ LIBNOTIFY_REQUIRED=0.4.4
 LIBCANBERRA_GTK_REQUIRED=0.4
 LIBCHAMPLAIN_REQUIRED=0.3.3
 LIBCHAMPLAIN_GTK_REQUIRED=0.3.3
-CLUTTER_GTK_REQUIRED=0.8.2
+CLUTTER_GTK_REQUIRED=0.10.0
 GEOCLUE_REQUIRED=0.11
 WEBKIT_REQUIRED=1.1.7
 NETWORK_MANAGER_REQUIRED=0.7.0
@@ -259,7 +259,7 @@ if test "x$enable_map" != "xno"; then
    [
       champlain-0.3 >= $LIBCHAMPLAIN_REQUIRED,
       champlain-gtk-0.3 >= $LIBCHAMPLAIN_GTK_REQUIRED
-      clutter-gtk-0.8 >= $CLUTTER_GTK_REQUIRED
+      clutter-gtk-0.10 >= $CLUTTER_GTK_REQUIRED
    ], have_libchamplain="yes", have_libchamplain="no")
 
    if test "x$have_libchamplain" = "xyes"; then
diff --git a/src/empathy-map-view.c b/src/empathy-map-view.c
index 86e15bd..3bcc5f3 100644
--- a/src/empathy-map-view.c
+++ b/src/empathy-map-view.c
@@ -26,7 +26,7 @@
 
 #include <champlain/champlain.h>
 #include <champlain-gtk/champlain-gtk.h>
-#include <clutter-gtk/gtk-clutter-embed.h>
+#include <clutter-gtk/clutter-gtk.h>
 #include <telepathy-glib/util.h>
 
 #include <libempathy/empathy-contact.h>
@@ -190,7 +190,7 @@ map_view_contacts_foreach (GtkTreeModel *model,
   if (avatar != NULL)
     {
       texture = clutter_texture_new ();
-      gtk_clutter_texture_set_from_pixbuf (CLUTTER_TEXTURE (texture), avatar);
+      gtk_clutter_texture_set_from_pixbuf (CLUTTER_TEXTURE (texture), avatar, NULL);
       champlain_marker_set_image (CHAMPLAIN_MARKER (marker), texture);
       g_object_unref (avatar);
     }
diff --git a/src/empathy.c b/src/empathy.c
index 4ff0918..15155d2 100644
--- a/src/empathy.c
+++ b/src/empathy.c
@@ -32,7 +32,7 @@
 #include <gdk/gdkx.h>
 
 #if HAVE_LIBCHAMPLAIN
-#include <clutter-gtk/gtk-clutter-embed.h>
+#include <clutter-gtk/clutter-gtk.h>
 #endif
 
 #include <libebook/e-book.h>
